crates/typst-syntax/src/file.rs

```diff
@@ -213,25 +213,25 @@ impl FromStr for PackageSpec {
 
         let namespace = s.eat_until('/');
         if namespace.is_empty() {
-            Err("package specification is missing namespace")?;
+            Err("package specification \"@{namespace}/{name}:{version}\" is missing namespace")?;
         } else if !is_ident(namespace) {
-            Err(eco_format!("`{namespace}` is not a valid package namespace"))?;
+            Err(eco_format!("`{namespace}` is not a valid package namespace; expected \"@{{namespace}}/{{name}}:{{version}}\""))?;
         }
 
         s.eat_if('/');
 
         let name = s.eat_until(':');
         if name.is_empty() {
-            Err("package specification is missing name")?;
+            Err("package specification \"@{namespace}/{name}:{version}\" is missing name")?;
         } else if !is_ident(name) {
-            Err(eco_format!("`{name}` is not a valid package name"))?;
+            Err(eco_format!("`{name}` is not a valid package name; expected \"@{{namespace}}/{{name}}:{{version}}\""))?;
         }
 
         s.eat_if(':');
 
         let version = s.after();
         if version.is_empty() {
-            Err("package specification is missing version")?;
+            Err("package specification \"@{namespace}/{name}:{version}\" is missing version")?;
         }
```

Comment:
You have to adjust the test cases for those, make sure to run `cargo test --workspace` to see how to A) update them if they change, B) if all your changes retained the original behavior.